roxie/ccd/ccdcontext.cpp

```diff
@@ -429,15 +429,23 @@ class CRoxieWorkflowMachine : public WorkflowMachine
 
     inline bool fileExists(const char *lfn)
     {
-        Owned<IDistributedFile> f = queryDistributedFileDirectory().lookup(lfn, NULL);  // MORE - need a userdescriptor from workunit
+        Owned<IDistributedFile> f = queryDistributedFileDirectory().lookup(lfn, queryUserDescriptor());
         if (f)
             return true;
         return false;
     }
 
     inline IUserDescriptor *queryUserDescriptor()
     {
-        return workunit->queryUserDescriptor();
+        if (workunit)
+            return workunit->queryUserDescriptor();//ad-hoc mode
```

Comment:
@richardkchapman these changes look ok, but the old code seemed to assume that this class always had a workunit... was that a bug? Or perhaps it worked in the scenarios this method was being called?

If there is always a workunit in this class the changes may not be needed or if there is no workunit it might be treated as an error.